core/capabilities/ccip/launcher/diff.go

```diff
@@ -113,28 +104,22 @@ func filterCCIPDONs(
 // checkCapabilityPresence checks if the capability with the given version and
 // labelled name is present in the given capability registry state.
 func checkCapabilityPresence(
-	capabilityVersion,
-	capabilityLabelledName string,
-	state registrysyncer.State,
-) (kcr.CapabilitiesRegistryCapabilityInfo, error) {
+	capabilityID string,
```

Comment:
Feels less error prone to make this structured ID an actual structure rather than just a `string`. 

@cedric-cordenier wdyt? Maybe something like:

```golang
type CapabilityID struct {
    Name string
    Version string
}

func (cid CapabilityID) String() string {
    // perhaps some semver regex validation using the semver package
    return fmt.Sprintf("%s@%s", cid.Name, cid.Version)
}
```

Reply:
I said a similar comment on call with @cedric-cordenier last week as well :)

Reply:
This should not be a blocker for this PR though.

Reply:
@asoliman92 can you create a ticket in the backlog for this if you're not going to do it in this PR

Reply:
Done